contrib/test.sh

```diff
@@ -69,10 +69,10 @@ then
     )
 fi
 
-# Bench if told to
+# Bench if told to (only works with nightly toolchain).
 if [ "$DO_BENCH" = true ]
 then
-    cargo bench --features unstable
+    RUSTFLAGS='--cfg=bench' cargo bench
```

Comment:
Maybe `cargo +nightly bench`?

Reply:
This would make it impossible to use specific toolchains for benching, no?

Reply:
If you have multiple nightlies, yes (is that even possible?) Also I think `+beta` is also possible. Stable can't work because `extern crate test` is experimental.

Reply:
You can definitely have multiple nightlies; you specify them by date. This is sometimes useful for bisecting when a nightly broke things.

Understood that `stable` won't work...we should, I suppose, check whether the user has specified a toolchain and provide a helpful error message if he hasn't.

Reply:
By specify you mean by using `TOOLCHAIN=foo`, right? We don't do that in `test.sh` so we can't warn for not being set.

Reply:
That is what I mean, and I don't understand what you mean by "we don't do that". We have a check for it on line 6.

Reply:
Oh I mean we never set the `TOOLCHAIN` env variable in CI. We could check if `TOOLCHAIN` is set, and check what toolchain is being used, then output an error message if `TOOLCHAIN` is set and current toolchain is not nightly.